bayesian_cnn_prometheus/analysis/masks_analyzer.py

```diff
@@ -0,0 +1,63 @@
+import json
+import os
+import numpy as np
+
+from bayesian_cnn_prometheus.analysis.similarity_comparer import SimilarityComparer
+from typing import List
+
+
+class MasksAnalyzer:
+    def __init__(self, model_name: str, lesion_masks_path: str, variance_masks_path: str):
+        """
+        Creates MasksAnalyzer object to compare generated variance masks with respective lesion masks.
+        :param model_name: name of the model that has generated variance masks
+        :param lesion_masks_path: path to a file with original lesion masks
+        :param variance_masks_path: path to a file with generated variance masks
+        """
+        self.model_name = model_name
+        self.lesion_masks_path = lesion_masks_path
+        self.variance_masks_path = variance_masks_path
+        self.lesion_mask_names = self.load_mask_names(self.lesion_masks_path)
+        self.variance_mask_names = self.load_mask_names(self.variance_masks_path)
+        self.metrics = {
+            "dice_coefficient": [],
+            "hausdorff_distance": [],
+            "jaccard_index": [],
+        }
+
+    def perform_analysis(self, save_to_json: bool = False):
+        """
+        Assigns similarity metrics describing each lesion and variance masks pair.
+        :param save_to_json: if True, saves mean of corresponding metrics to JSON file
+        """
+        for lesion_mask_name, variance_mask_name in zip(self.lesion_mask_names, self.variance_mask_names):
+            path_to_lesion_mask = self.lesion_masks_path + lesion_mask_name
+            path_to_variance_mask = self.variance_masks_path + variance_mask_name
+            similarity_comparer = SimilarityComparer(path_to_lesion_mask, path_to_variance_mask)
+            similarity_comparer.perform_analysis()
+            metrics = similarity_comparer.get_metrics()
+
+            if metrics:
+                for key in metrics:
+                    self.metrics[key].append(metrics[key])
+
+        if save_to_json:
+            self._save_to_json()
+
+    @staticmethod
+    def load_mask_names(path_to_masks: str) -> List[str]:
+        """
+        Loads mask paths from specified path.
+        :param path_to_masks: a path that mask paths will be loaded from
```

Comment:
Hmm może "path to dir with mask files" ?

Reply:
Racja, będzie bardziej opisowe